lib/phoenix_live_view.ex

```diff
@@ -1322,4 +1374,87 @@ defmodule Phoenix.LiveView do
   end
 
   defp child?(%Socket{parent_pid: pid}), do: is_pid(pid)
+
+  @doc """
+  Attaches the given `fun` by `name` for the lifecycle `stage` into `socket`.
+
+  > Note: This function is for server-side lifecycle callbacks.
+  > For client-side hooks, see the
+  > [JS Interop guide](js-interop.html#client-hooks).
+
+  Hooks provide a mechanism to tap into key stages of the LiveView
+  lifecycle in order to bind/update assigns, intercept events,
+  patches, and regular messages when necessary, and to inject
+  common functionality. Hooks may be attached to any of the following
+  lifecycle stages: `:mount` (via `on_mount/1`), `:handle_params`,
+  `:handle_event`, and `:handle_info`.
+
+  ## Return Values
+
+  Lifecycle hooks are effectively a single
+  [`reduce_while`](`Enum.reduce_while/3`) operation that takes place
+  immediately before a given lifecycle callback is invoked on the
+  LiveView. A hook may return `{:halt, socket}` to halt the reduction,
+  otherwise it must return `{:cont, socket}` so the operation may
+  continue until all hooks have been invoked for the current stage.
+
+  ## Halting the lifecycle
+
+  Note that halting from a hook _will halt the entire lifecycle stage_.
+  This means that when a hook returns `{:halt, socket}` then the
+  LiveView callback will **not** be invoked. This has some
+  implications.
+
+  ### Implications for plugin authors
+
+  When defining your plugin, you **must** define a catch-all clause
+  that either halts or continues _and_ you should make this behaviour
+  clear to the end-user, perhaps exposing it as a config option if
+  sensible.
+
+  ### Implications for end-users
+
+  Let's say you invoke a plugin that attaches a hook for a callback
+  that you did not define. You are not required to define a
+  catch-all for that callback in your LiveView.
```

Comment:
I am usually not a fan of saying what you don't have to do. Because, if you don't have to do something, why talk about it? So I think we may be able to get rid of this paragraph?